mlem/api/commands.py

```diff
@@ -85,7 +86,7 @@ def apply(
         resolved_method = PREDICT_METHOD_NAME
     echo(EMOJI_APPLY + f"Applying `{resolved_method}` method...")
     res = [
-        w.call_method(resolved_method, get_dataset_value(part))
+        w.call_method(resolved_method, get_dataset_value(part, batch))
```

Comment:
Here you should flatten all the batches if there are any. Here the example of how I see it:
Suppose you have `dataframe=[1,2,3,4]` (I mean 1 column, 4 rows) saved to csv file. You load its metadata without loading the value, let's say `dt = DatasetMeta(dataset_type=DataFrameType(...), ...)`. And you call `apply` with data=[dt]. If batch arg is not provided, what will happen is `get_dataset_value` will load the actual dataframe and in the end `res = [w.call_method(..., dataframe([1,2,3,4))]`. But if you provided `batch=2`, `dt. read_batch` should be called. If you iterate through it, you will get 2 parts of the dataframe, and in the end `res=[w.call_method(..., dataframe([1,2])), w.call_method(..., dataframe([3,4]]))]`

Reply:
👍🏻 Makes sense, I implemented something similar yesterday, but I couldn't sync the changes to this PR because the private fork no longer points to this repository.